app/features/database.py

```diff
@@ -0,0 +1,88 @@
+import aiosqlite
+from interactions import Extension, listen
+from interactions.api import events
+
+
+class Database(Extension):
+    """extension class adding access to a database connection via a db attribute of the bot instance."""
+
+    async def async_start(self) -> None:
+        """Connect to db as bot loops starts."""
+        self.bot.db = self
+        self.bot.db_conn = await aiosqlite.connect("./ee.db")
+        await self.populate_tables()
+
+    async def populate_tables(self) -> None:
+        """Run the coroutines to create the db tables."""
+        await self.todo_table()
+
+    @listen(events.Connect)
+    async def bot_connect(self, event: events.Connect) -> None:
+        """Reconnect to db when the bot reconnects."""
+        print(event)
+        if not self.bot.db_conn:
+            self.bot.db_conn = await aiosqlite.connect("./ee.db")
+
+    @listen(events.Disconnect)
+    async def bot_disconnect(self, event: events.Disconnect) -> None:
+        """Disconnect the db when the bot disconnects."""
+        print(event)
+        await self.bot.db_conn.close()
+        self.bot.db = None
+
+    # TO-DO methods
+
+    async def todo_table(self) -> None:
+        """Creation of to-do db table."""
+        async with self.bot.db_conn.cursor() as cursor:
+            await cursor.execute("""
+            CREATE TABLE IF NOT EXISTS
+            todo (item_id INTEGER PRIMARY KEY AUTOINCREMENT,
```

Comment:
Currently with the primary key it doesn't quite work like I intended to access the items.
When I show the list to the user I get all the items and then numerate them beginning at 1.
If I were to remove item with id=3 in a list of 10 items, then in the database this id would be deleted. However the new list still has 9 items, so numerated from 1 to 9. I wanted to remove the 'new' 3rd item the removal won't work because the database has no more item with id=3

Reply:
So would need to hold the db id internally separate from the enumerated one that's visible to the user, or alternatively delete based on item string itself; though as its not unique (unless we make it so) could remove a random/first item.

Reply:
If deleting from the string is easier, then lets do that. And we can also make the string unique.